octoprint_mrbeam/gcodegenerator/jobtimeestimation.py

```diff
@@ -17,6 +17,35 @@
 FIND_X_VALUE = r"X(\d+\.?\d+)"
 FIND_Y_VALUE = r"Y(\d+\.?\d+)"
 FIND_FEEDRATE = r"F(\d+\.?\d+)"
+MATCH_COMMENT_ADD_TIME = re.compile(r"[\+-]?[0-9]+\.?[0-9]*s")
+
+
+def time_from_comment(comment):
+    """
+    Returns the number of seconds read in the comment if the comment matches a float:
+        ; [\+-]?[0-9]+\.?[0-9]*s
+
+    Examples:
+        >>> time_from_comment("0.19s")
+        0.19
+        >>> time_from_comment("; +0.19s")
+        0.19
+        >>> time_from_comment("; +50s")
+        50.0
+        >>> time_from_comment("; this is not a number")
+        0.0
+        >>> time_from_comment("; here is a number -70s")
+        -70.0
+        >>> time_from_comment("; not formatted 70")
+        0.0
+        >>> time_from_comment("cannot do 2 numbers 6.9s 42.0s")
+        6.9
+    """
+    match = MATCH_COMMENT_ADD_TIME.search(comment)
+    if match:
+        float(match.group(0))
+    else:
+        return 0.0
```

Comment:
Can we have this more explicit with a key word eg: `EXTRA_JOB_TIME:45.2`
This way  it's more clear that there is a system that parses this comment and it prevents humans from adding job time by accident.

I appreciate the documentation though

Reply:
Note to self : It won't parse succesfully if the "s" is added -> `group(1)` and change regex